scavenger-api/src/main/kotlin/com/navercorp/scavenger/service/SnapshotNodeService.kt

```diff
@@ -112,40 +113,40 @@ class SnapshotNodeService(
     * "a.b.c.d(e, f)" to ["a", "b", "c", "d(e, f)"]
     * "a.b.c.D(e, f)" to ["a", "b", "c", "D", "D(e, f)"]
     */
-    private fun splitSignature(methodInvocationEntity: MethodInvocationEntity): List<String> {
+    private fun splitSignature(methodInvocationEntity: MethodInvocationEntity): List<Map<String, String>> {
         val (nameOnlySignature, arguments) = methodInvocationEntity.signature.split("(", limit = 2)
 
-        val elements = nameOnlySignature.split(*DELIMITERS)
+        val elements = nameOnlySignature.split("(?=\\b[.$])".toRegex())
             .filterNot { it == "" }
+            .map { mutableMapOf("signature" to it.replace(".", "")) }
             .toMutableList()
 
         if (isConstructor(methodInvocationEntity)) {
-            elements.add(elements.last())
+            val constructorSignature = checkNotNull(elements.last()["signature"]) { "signature must be not null" }
+            elements.add(mutableMapOf("signature" to constructorSignature.replace("$", "")))
+        }
+
+        val lastElement = elements.last()
+        lastElement["signature"] = "${lastElement["signature"]}($arguments" + if (arguments.last() != ')') ")" else ""
+
+        elements.asReversed().forEachIndexed { index, mutableMap ->
+            val signature = checkNotNull(mutableMap["signature"]) { "signature must be not null" }
+            mutableMap["type"] = if (signature.contains("(")) {
+                Node.Type.METHOD.name
+            } else if (signature.contains("$") || checkNotNull(elements[elements.size - index]["signature"]).contains("[($]".toRegex())) {
```

Comment:
How about removing `asReversed()` and just doing `elements(index + 1)`?

Reply:
The reason I used as `Reversed()` was that I thought we had to go in reverse order to distinguish between `package` and `classe`.
Before, if the first letter is a capital letter or parent type is a `class`, we thought it was a `class`.
Now we can't use the first one, so I don't think there's a way to distinguish between `package` and `class` if we go in order.

Reply:
Yes, you're right, I thought that might be something that could be checked at `elements[index +1]["signature"]).contains("[($]".toRegex())`.

Reply:
um....
if we use `asReversed()` and `forEachIndexed` together, the index is not in reverse order. It starts from 0....

Reply:
I had missed it. If we search in order as you said, we can distinguish between `package` and `class`.
I'll fix it.